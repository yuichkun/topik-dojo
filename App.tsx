/**
 * TOPIK道場 - React Native App
 * トップ画面（級選択）
 */

import React, { useEffect, useState } from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
  StatusBar,
  useColorScheme,
  Alert,
  SafeAreaView,
} from 'react-native';

interface TopScreenProps {}

const TopScreen: React.FC<TopScreenProps> = () => {
  const [reviewCount, setReviewCount] = useState<number>(0);
  const isDarkMode = useColorScheme() === 'dark';

  // 復習対象数を取得（現在はモックデータ）
  useEffect(() => {
    const loadReviewCount = async () => {
      try {
        // TODO: 実際のDBから復習対象数を取得
        // 現在はモックデータを使用
        const mockReviewCount = 15;
        setReviewCount(mockReviewCount);
        
        // TODO: アプリアイコンバッジの更新
        // setBadgeCount(mockReviewCount);
      } catch (error) {
        console.error('復習対象数の取得に失敗:', error);
        // エラーの場合は0件として扱う
        setReviewCount(0);
      }
    };

    loadReviewCount();
  }, []);

  // 復習ボタンのタップハンドラ
  const handleReviewPress = () => {
    if (reviewCount === 0) {
      return; // 復習対象が0件の場合は何もしない
    }
    
    // TODO: 復習画面への遷移
    Alert.alert('復習画面', `${reviewCount}語の復習を開始します`);
  };

  // 級選択ボタンのタップハンドラ
  const handleLevelPress = (level: number) => {
    // TODO: 学習モード選択画面への遷移
    Alert.alert('級選択', `${level}級の学習モードを選択してください`);
  };

  const levels = [1, 2, 3, 4, 5, 6];

  return (
    <SafeAreaView style={styles.container}>
      <StatusBar 
        barStyle={isDarkMode ? 'light-content' : 'dark-content'}
        backgroundColor="#ffffff"
      />
      
      {/* アプリタイトル */}
      <View style={styles.header}>
        <Text style={styles.title}>TOPIK道場</Text>
      </View>

      {/* 復習ボタン */}
      <View style={styles.reviewSection}>
        <TouchableOpacity
          style={[
            styles.reviewButton,
            reviewCount === 0 && styles.reviewButtonDisabled
          ]}
          onPress={handleReviewPress}
          disabled={reviewCount === 0}
        >
          <Text style={[
            styles.reviewButtonText,
            reviewCount === 0 && styles.reviewButtonTextDisabled
          ]}>
            復習 ({reviewCount}語)
          </Text>
        </TouchableOpacity>
      </View>

      {/* 級選択ボタン */}
      <View style={styles.levelSection}>
        <View style={styles.levelGrid}>
          {/* 上段: 1級, 2級, 3級 */}
          <View style={styles.levelRow}>
            {levels.slice(0, 3).map((level) => (
              <TouchableOpacity
                key={level}
                style={styles.levelButton}
                onPress={() => handleLevelPress(level)}
              >
                <Text style={styles.levelButtonText}>{level}級</Text>
              </TouchableOpacity>
            ))}
          </View>
          
          {/* 下段: 4級, 5級, 6級 */}
          <View style={styles.levelRow}>
            {levels.slice(3, 6).map((level) => (
              <TouchableOpacity
                key={level}
                style={styles.levelButton}
                onPress={() => handleLevelPress(level)}
              >
                <Text style={styles.levelButtonText}>{level}級</Text>
              </TouchableOpacity>
            ))}
          </View>
        </View>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#ffffff',
  },
  header: {
    alignItems: 'center',
    paddingTop: 60,
    paddingBottom: 40,
  },
  title: {
    fontSize: 32,
    fontWeight: 'bold',
    color: '#333333',
    letterSpacing: 2,
  },
  reviewSection: {
    alignItems: 'center',
    paddingVertical: 30,
  },
  reviewButton: {
    borderWidth: 2,
    borderColor: '#007AFF',
    borderRadius: 8,
    paddingHorizontal: 40,
    paddingVertical: 15,
    backgroundColor: '#ffffff',
  },
  reviewButtonDisabled: {
    borderColor: '#CCCCCC',
    backgroundColor: '#F5F5F5',
  },
  reviewButtonText: {
    fontSize: 18,
    fontWeight: '600',
    color: '#007AFF',
  },
  reviewButtonTextDisabled: {
    color: '#999999',
  },
  levelSection: {
    flex: 1,
    justifyContent: 'center',
    paddingHorizontal: 20,
  },
  levelGrid: {
    alignItems: 'center',
  },
  levelRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    width: '100%',
    marginVertical: 15,
  },
  levelButton: {
    flex: 1,
    borderWidth: 1,
    borderColor: '#333333',
    borderRadius: 4,
    paddingVertical: 20,
    marginHorizontal: 10,
    alignItems: 'center',
    backgroundColor: '#ffffff',
  },
  levelButtonText: {
    fontSize: 16,
    fontWeight: '500',
    color: '#333333',
  },
});

export default TopScreen;
