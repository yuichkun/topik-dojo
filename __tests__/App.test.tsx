/**
 * @format
 */

import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import App from '../App';

describe('App', () => {
  test('renders correctly', async () => {
    let component: ReactTestRenderer.ReactTestRenderer;
    
    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<App />);
    });
    
    expect(component!).toBeDefined();
  });

  test('matches snapshot', async () => {
    let component: ReactTestRenderer.ReactTestRenderer;
    
    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<App />);
    });
    
    expect(component!.toJSON()).toMatchSnapshot();
  });
});
