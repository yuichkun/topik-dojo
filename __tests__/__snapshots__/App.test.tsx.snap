// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      activityState={2}
      aria-hidden={false}
      collapsable={false}
      fullScreenSwipeShadowEnabled={true}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      hasLargeHeader={false}
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onHeaderHeightChange={[Function]}
      onNativeDismissCancelled={[Function]}
      onSheetDetentChanged={[Function]}
      onTransitionProgress={[Function]}
      onWillAppear={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents={
        [
          1,
        ]
      }
      sheetCornerRadius={-1}
      sheetElevation={24}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetInitialDetent={0}
      sheetInitialDetentIndex={0}
      sheetLargestUndimmedDetent={-1}
      sheetLargestUndimmedDetentIndex={-1}
      stackPresentation="push"
      style={
        [
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ],
          {
            "zIndex": undefined,
          },
        ]
      }
      swipeDirection="horizontal"
    >
      <RNSScreenContentWrapper
        collapsable={false}
        style={
          [
            {
              "flex": 1,
            },
            [
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ],
          ]
        }
      >
        <RCTSafeAreaView
          className="flex-1 bg-white"
        >
          <View
            className="items-center pt-16 pb-10"
          >
            <Text
              className="text-3xl font-bold text-gray-800 tracking-wider"
            >
              TOPIK道場
            </Text>
          </View>
          <View
            className="items-center py-8"
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": true,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                className="
            text-lg font-semibold
            text-gray-400
          "
              >
                復習 (
                0
                語)
              </Text>
            </View>
          </View>
          <View
            className="flex-1 justify-center px-5"
          >
            <View
              className="items-center"
            >
              <View
                className="flex-row justify-between w-full my-4"
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    className="text-base font-medium text-gray-800"
                  >
                    1
                    級
                  </Text>
                </View>
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    className="text-base font-medium text-gray-800"
                  >
                    2
                    級
                  </Text>
                </View>
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    className="text-base font-medium text-gray-800"
                  >
                    3
                    級
                  </Text>
                </View>
              </View>
              <View
                className="flex-row justify-between w-full my-4"
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    className="text-base font-medium text-gray-800"
                  >
                    4
                    級
                  </Text>
                </View>
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    className="text-base font-medium text-gray-800"
                  >
                    5
                    級
                  </Text>
                </View>
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    className="text-base font-medium text-gray-800"
                  >
                    6
                    級
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </RCTSafeAreaView>
      </RNSScreenContentWrapper>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleFontFamily="System"
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleFontFamily="System"
        largeTitleFontWeight="700"
        largeTitleHideShadow={false}
        pointerEvents="box-none"
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "position": "absolute",
            "width": "100%",
          }
        }
        title="Top"
        titleColor="rgb(28, 28, 30)"
        titleFontFamily="System"
        titleFontWeight="600"
        topInsetEnabled={true}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
